x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_definition_section.tsx

```diff
@@ -89,16 +89,21 @@ export const Filters = ({
   index,
   'data-test-subj': dataTestSubj,
 }: FiltersProps) => {
-  const flattenedFilters = mapAndFlattenFilters(filters);
-
   const defaultIndexPattern = useDefaultIndexPattern();
-
-  const { indexPattern } = useRuleIndexPattern({
-    dataSourceType: dataViewId ? DataSourceType.DataView : DataSourceType.IndexPatterns,
-    index: index ?? defaultIndexPattern,
-    dataViewId,
-  });
-
+  const useDataViewParams = dataViewId
+    ? { dataViewId }
+    : { indexPatterns: index ?? defaultIndexPattern };
+  const { dataView } = useDataView(useDataViewParams);
+  const isEsql = filters.some((filter) => filter?.query?.language === 'esql');
+  const searchBarFilters = useMemo(() => {
+    if (!index || isDataView(index) || isEsql) {
+      return filters;
+    }
+    const updatedFilters = cloneDeep(filters);
+    updatedFilters.forEach((filter) => (filter.meta.index = index.join(',')));
+    return updatedFilters;
```

Comment:
```suggestion
    const filtersWithMetaIndex = filters.map((filter) => {
      return {
        ...filter,
        meta: {
          ...filter.meta,
          index: index.join(','),
        },
      };
    });

    return filtersWithMetaIndex;
```
Any reason why you are cloning the filters, and then editing the meta object in place? Look anti-pattern-ny to me. I would rather use the immutable approach suggested above.

Reply:
Ok, I will improve, thank you.